lib/most_frequent_integer.rb

```diff
@@ -3,5 +3,28 @@
 # exists, return nil.
 
 def find_most_frequent_integer(arr)
-  # put your code here
+  #use a Hash to keep track of the number of duplicates
+  new_hash = Hash.new(0)
+  #iterate through each element in the array
+  arr.each do |idx|
+      #add all elements to the new_hash and add one to the key's value if there's a duplicate
+      new_hash[idx] += 1
+  end
+  #make sure that the Array passed in contained something
+  if new_hash.empty?
+      return nil
+  end
+  #----------------------------------------------------------------
+  #Do more iteration to check if one value is greater than the rest??? 
+  #----------------------------------------------------------------
+  #iterate through each key and value in new_hash
+  new_hash.each do |x, y|
+      #if the value is greater than the key then that key had duplicates in the original Array
+      if y > x
```

Comment:
Okay, I think the issue might be here. 

Let's say you start with the array: `[1, 2, 2, 3]`
After the block on lines 9-12, you have a hash: 
```ruby
{1 => 1,
 2 => 2,
 3 => 1}
```

Now in this loop, you are checking to see if the `y > x` or if, for every pair in the hash,  the _value_ is greater than the _key_.

Issues with this:

1) Right away, we can evaluate `if 1 > 1` and we know it to be `false`. Based on this conditional, the entire function would return `nil` (line 27), instead of a number which is what we'd like the return to be.

2) Say we **did** get to evaluating the key, value pair `2 => 2`. We know from looking at the original array that this is the most freq. integer, and we want that _key_ to be returned. But, again, the expression `2 > 2` evaluates to `false`.  So I don't think that comparing the _key_ (which represents the _integer_ we are evaluating) and the _value_ (the _count_ of that integer) is helpful. 

In order to determine which _integer_ has the highest _count_, what parts of the hash should we be comparing?